cmd/vpn_client/app/pathcontroller/ping.go

```diff
@@ -77,14 +90,126 @@ func (p icmpPinger) ping(client net.IP) error {
 		return fmt.Errorf("error reading response: %w", err)
 
 	}
-	rm, err := icmp.ParseMessage(protocolICMP, rb[:n])
+	rm, err := icmp.ParseMessage(ipv6.ICMPTypeEchoReply.Protocol(), rb[:n])
 	if err != nil {
 		return fmt.Errorf("error parsing response: %w", err)
 	}
 	switch rm.Type {
-	case ipv4.ICMPTypeEchoReply:
+	case ipv6.ICMPTypeEchoReply:
 		return nil
 	default:
 		return err
 	}
 }
+
+func (p icmpPinger) neighborSolicitation(client net.IP) error {
+	if len(client) != net.IPv6len {
+		return fmt.Errorf("only usable with ipv6")
+	}
+	ifi, err := net.InterfaceByName("bond0")
+	if err != nil {
+		return fmt.Errorf("bonding device not found: %w", err)
+	}
+	ips, err := network.GetLinkIPAddressesByName("bond0", network.ScopeLink)
+	if err != nil {
+		return fmt.Errorf("getting link IP address failed: %w", err)
+	}
+	if len(ips) != 1 {
+		return fmt.Errorf("link IP address not unique: %d", len(ips))
+	}
+
+	ns := icmp.Message{
+		Type: ipv6.ICMPTypeNeighborSolicitation,
+		Code: 0,
+		Body: &icmp.RawBody{
+			Data: buildNSBody(client, ifi.HardwareAddr),
+		},
+	}
+
+	data, err := ns.Marshal(nil)
+	if err != nil {
+		return fmt.Errorf("error marshalling ICMP message: %w", err)
+	}
+
+	conn, err := icmp.ListenPacket("ip6:ipv6-icmp", ips[0].String()+"%bond0")
+	if err != nil {
+		return fmt.Errorf("error opening raw socket: %w", err)
+	}
+	defer conn.Close()
+
+	deadline := time.Now().Add(p.timeout / 2)
+	pc := conn.IPv6PacketConn()
+	if err := pc.SetReadDeadline(deadline); err != nil {
+		return fmt.Errorf("error setting deadline: %w", err)
+	}
+	if err := pc.SetHopLimit(255); err != nil {
+		return fmt.Errorf("error setting hop limit: %w", err)
+	}
+	if err := pc.SetMulticastHopLimit(255); err != nil {
+		return fmt.Errorf("error setting hop limit: %w", err)
+	}
+
+	// calculate solicited-node multicase address
+	destIP := net.ParseIP("ff02::1:ff00:0")
+	copy(destIP[13:], client[13:]) // copy last 24 bits
+	dst := &net.IPAddr{IP: destIP}
+	_, err = pc.WriteTo(data, nil, dst)
+	if err != nil {
+		return fmt.Errorf("error sending ICMP message: %w", err)
+	}
+
+	// Listen for Neighbor Advertisement response
+	reply := make([]byte, 1500)
+	n, _, _, err := pc.ReadFrom(reply)
+	if err != nil {
+		if strings.Contains(err.Error(), "i/o timeout") {
+			return fmt.Errorf("i/o timeout")
```

Comment:
Because it spams the log with lengthy and boring details. As the timeout happens after in a longer time period between the kube-apiserver pods have started and the VPN tunnel is up, the logs are kept concise.